src/jit/hwintrinsicxarch.cpp

```diff
@@ -152,6 +157,98 @@ InstructionSet Compiler::isaOfHWIntrinsic(NamedIntrinsic intrinsic)
     return hwIntrinsicInfoArray[intrinsic - NI_HW_INTRINSIC_START - 1].isa;
 }
 
+//------------------------------------------------------------------------
+// ivalOfHWIntrinsic: get the imm8 value of this intrinsic from the hwIntrinsicInfoArray table
+//
+// Arguments:
+//    intrinsic -- id of the intrinsic function.
+//
+// Return Value:
+//     the imm8 value of the intrinsic, -1 for non-IMM intrinsics
+//
+int Compiler::ivalOfHWIntrinsic(NamedIntrinsic intrinsic)
+{
+    assert(intrinsic != NI_Illegal);
+    assert(intrinsic > NI_HW_INTRINSIC_START && intrinsic < NI_HW_INTRINSIC_END);
+    return hwIntrinsicInfoArray[intrinsic - NI_HW_INTRINSIC_START - 1].ival;
+}
+
+//------------------------------------------------------------------------
+// simdSizeOfHWIntrinsic: get the SIMD size of this intrinsic
+//
+// Arguments:
+//    intrinsic -- id of the intrinsic function.
+//
+// Return Value:
+//     the SIMD size of this intrinsic
+//         - from the hwIntrinsicInfoArray table if intrinsic is NOT HW_Category_UnfixedSIMDSize
+//         - TODO - from the signature if intrinsic is HW_Category_UnfixedSIMDSize
+//
+// Note - this function is only used by the importer
+//        after importation (i.e., codegen), we can get the SIMD size from GenTreeHWIntrinsic IR
+static unsigned simdSizeOfHWIntrinsic(NamedIntrinsic intrinsic, CORINFO_SIG_INFO* sig)
+{
+    assert(intrinsic > NI_HW_INTRINSIC_START && intrinsic < NI_HW_INTRINSIC_END);
+    return hwIntrinsicInfoArray[intrinsic - NI_HW_INTRINSIC_START - 1].simdSize;
+}
+
+//------------------------------------------------------------------------
+// insOfHWIntrinsic: get the instruction of the given intrinsic
+//
+// Arguments:
+//    intrinsic -- id of the intrinsic function.
+//    type      -- vector base type of this intrinsic
+//
+// Return Value:
+//     the instruction of the given intrinsic on the base type
+//     return INS_invalid for unsupported base types
+//
+instruction Compiler::insOfHWIntrinsic(NamedIntrinsic intrinsic, var_types type)
+{
+    assert(intrinsic != NI_Illegal);
+    assert(intrinsic > NI_HW_INTRINSIC_START && intrinsic < NI_HW_INTRINSIC_END);
+    assert(type >= TYP_BYTE && type <= TYP_DOUBLE);
```

Comment:
Would still be good to document that this is relying on the current ording of the typlist.

Reply:
The current code base heavily relies in the order https://github.com/dotnet/coreclr/blob/master/src/jit/vartype.h#L197